dipy/denoise/noise_estimate.py

```diff
@@ -203,15 +188,21 @@ def piesno_3D(data, N, alpha=0.01, l=100, itermax=100, eps=1e-5):
     else:
         q = 0.5
 
-    # prevent overflow in sum_m2
-    data = data.astype(np.float32)
-
     # Initial estimation of sigma
     denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
     m = np.percentile(data, q * 100) / denom
+
+    # if the percentile is 0, then more than half of the slice is zero and give up
+    if m == 0:
```

Comment:
Please take a look at #576, to make sure everything made it there. It was a
bit of a tricky rebase.

On Mon, Feb 23, 2015 at 12:20 PM, Ariel Rokem <arokem@gmail.com> wrote:

> OK. I am going to go ahead and merge this into my PR, and then do some
> more work there to prepare this for another round of reviews, and
> hopefully, at long last, a merge.
>
> On Mon, Feb 23, 2015 at 12:12 PM, Samuel St-Jean <notifications@github.com
> > wrote:
>
>> In dipy/denoise/noise_estimate.py
>> <https://github.com/arokem/dipy/pull/13#discussion_r25197206>:
>>
>> >      # Initial estimation of sigma
>> >      denom = np.sqrt(2 * _inv_nchi_cdf(N, 1, q))
>> >      m = np.percentile(data, q * 100) / denom
>> > +
>> > +    # if the percentile is 0, then more than half of the slice is zero and give up
>> > +    if m == 0:
>>
>> Well, the first estimate is based on a quantile, starting from ~0.79 to
>> the median (0.5). If the first estimate by the quantile is 0, then at least
>> more than half of the voxels on the current slice are 0. In this case the
>> function simply exit with a warning.
>>
>> That being said, the computed thresholds afterward are going ot be really
>> low, but could still find stuff. I could change that for a sclice to be
>> exactly zero before exiting, and worse case scenario no voxel should be
>> found anyway.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/arokem/dipy/pull/13/files#r25197206>.
>>
>
>